tests/test_config/test_matcha_config.py

```diff
@@ -0,0 +1,243 @@
+
+    return MatchaConfigComponent(name="remote_state_bucket", properties=properties)
+
+
+@pytest.fixture
+def mocked_matcha_config(mocked_matcha_config_component: MatchaConfigComponent):
+    """A fixture returning a mocked MatchaConfig instance.
+
+    Args:
+        mocked_matcha_config_component (MatchaConfigComponent): a mocked MatchaConfigComponent instance.
+
+    Returns:
+        (MatchaConfig): a mocked MatchaConfig instance.
+    """
+    return MatchaConfig([mocked_matcha_config_component])
+
+
+@pytest.fixture
+def mocked_matcha_config_service():
+    """A fixture representing a mocked MatchaConfigService.
+
+    Returns:
+    (MatchaConfigService): a mocked MatchaConfigService instance.
+    """
+    return MatchaConfigService()
+
+
+def test_matcha_config_to_dict(
+    mocked_matcha_config_json_object: Dict[str, Dict[str, str]],
+    mocked_matcha_config: MatchaConfig,
+):
+    """Tests the to_dict method of the MatchaConfig class.
+
+    The test compares the output of the MatchaConfig to_dict method with a dictionary representation of the matcha.config.json file.
+
+    Args:
+        mocked_matcha_config_json_object (Dict[str, Dict[str, str]]): a dictionary representation of the matcha.config.json file
+        mocked_matcha_config (MatchaConfig): a mocked MatchaConfig instance.
+    """
+    assert mocked_matcha_config_json_object == mocked_matcha_config.to_dict()
+
+
+def test_matcha_config_from_dict(
+    mocked_matcha_config_json_object: Dict[str, Dict[str, str]],
+    mocked_matcha_config: MatchaConfig,
+):
+    """Tests the from_dict method of the MatchaConfig class.
+
+    The test compares the output of the MatchaConfig from_dict method with a mocked MatchaConfig instance.
+
+    Args:
+        mocked_matcha_config_json_object (Dict[str, Dict[str, str]]): a dictionary representation of the matcha.config.json file
+        mocked_matcha_config (MatchaConfig): a mocked MatchaConfig instance.
+    """
+    assert (
+        mocked_matcha_config.from_dict(mocked_matcha_config_json_object)
+        == mocked_matcha_config
+    )
+
+
+def test_matcha_config_service_write_matcha_config(
+    matcha_testing_directory: Iterator[str],
+    mocked_matcha_config_service: MatchaConfigService,
+    mocked_matcha_config: MatchaConfig,
+):
+    """Tests the write_matcha_config method of the MatchaConfigService.
+
+    The test creates a testing directory and a mock matcha.config.json file to test the read_matcha_config method.
+
+    Args:
+        matcha_testing_directory (Iterator[str]): a fixture for creating and removing temporary test directory for storing and moving files
+        mocked_matcha_config_service (MatchaConfigService): a mocked MatchaConfigService instance
+        mocked_matcha_config (MatchaConfig): a mocked MatchaConfig instance
+    """
+    matcha_config_file_path = os.path.join(
+        matcha_testing_directory, "matcha.config.json"
```

Comment:
Nit: `"matcha.config.json"` appears a lot in here, this should probably be a constant or a fixture

Reply:
Good point, updated.